library/src/main/kotlin/com/livefront/gsonkotlinadapter/KoltinReflectiveTypeAdapterFactory.kt

```diff
@@ -38,17 +41,30 @@ class KotlinReflectiveTypeAdapterFactory private constructor() : TypeAdapterFact
         require(!kotlinRawType.isSealed) { "Cannot serialize sealed class ${rawType.name}" }
         require(!kotlinRawType.isInner) { "Cannot serialize inner class ${rawType.name}" }
         kotlinRawType.primaryConstructor ?: return null
-        return Adapter(this, gson, type, kotlinRawType)
+        return Adapter(this, gson, type, kotlinRawType, enableDefaultPrimitiveValues)
     }
 
     internal class Adapter<T : Any>(
         factory: TypeAdapterFactory,
         gson: Gson,
         type: TypeToken<T>,
-        kClass: KClass<T>
+        kClass: KClass<T>,
+        private val enableDefaultPrimitiveValues: Boolean
     ) : TypeAdapter<T>() {
         private val primaryConstructor: KFunction<T> = kClass.primaryConstructor!!
         private val declaringClass: Class<T> = primaryConstructor.javaConstructor?.declaringClass!!
+        private val constructorParameterDefaultsMap: Map<KParameter, Any?> = primaryConstructor
+            .parameters
+            .map {
+                it to if (!it.isOptional) {
+                    if (enableDefaultPrimitiveValues) it.defaultPrimitiveValue else null
+                } else {
+                    Unit
+                }
+            }
+            .filter { it.second != Unit }
+            .toMap()
```

Comment:
I think I'd prefer if the `filter` came before the map, since then the `Unit` isn't used to overload `Any?`.

This should be equivalent:
```suggestion
            .filterNot(KParameter::isOptional)
            .associateWith { if (enableDefaultPrimitiveValues) it.defaultPrimitiveValue else null }
```

Reply:
Ah yeah I agree this seems better.

Reply:
Agreed